keel-core/src/main/kotlin/com/netflix/spinnaker/keel/validators/DeliveryConfigValidator.kt

```diff
@@ -0,0 +1,84 @@
+package com.netflix.spinnaker.keel.validators
+
+import com.netflix.spinnaker.keel.core.api.DependsOnConstraint
+import com.netflix.spinnaker.keel.core.api.SubmittedDeliveryConfig
+import com.netflix.spinnaker.keel.exceptions.DuplicateArtifactReferenceException
+import com.netflix.spinnaker.keel.exceptions.DuplicateResourceIdException
+import com.netflix.spinnaker.keel.exceptions.MissingEnvironmentReferenceException
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+/**
+ * Run validation checks against delivery config to ensure:
+ *
+ * - resources have unique ids
+ * - artifacts have unique references
+ * - depends on environments are unique
+ *
+ * Throws an exception if config fails any checks
```

Comment:
I think these comments belong in the function, not the class, especially since you mention the "throws" behavior. Here you could just say "Provides delivery config validation functions" or something like that.

Reply:
you are right!